app/sagas/init.js

```diff
@@ -29,71 +22,15 @@ export const initLocalSettings = function* initLocalSettings() {
 
 const restore = function* restore() {
 	try {
-		let hasMigration;
-		if (isIOS) {
-			hasMigration = yield AsyncStorage.getItem('hasMigration');
-		}
-
-		let { token, server } = yield all({
-			token: RNUserDefaults.get(RocketChat.TOKEN_KEY),
-			server: RNUserDefaults.get('currentServer')
+		const { token, server } = yield all({
+			token: UserPreferences.getStringAsync(RocketChat.TOKEN_KEY),
+			server: UserPreferences.getStringAsync(RocketChat.CURRENT_SERVER)
 		});
 
-		if (!hasMigration && isIOS) {
-			let servers = yield RNUserDefaults.objectForKey(SERVERS);
-			// if not have current
-			if (servers && servers.length !== 0 && (!token || !server)) {
-				server = servers[0][SERVER_URL];
-				token = servers[0][TOKEN];
-			}
-
-			// get native credentials
-			if (servers) {
-				try {
-					// parse servers
-					servers = yield Promise.all(servers.map(async(s) => {
-						await RNUserDefaults.set(`${ RocketChat.TOKEN_KEY }-${ s[SERVER_URL] }`, s[USER_ID]);
-						return ({ id: s[SERVER_URL], name: s[SERVER_NAME], iconURL: s[SERVER_ICON] });
-					}));
-					const serversDB = database.servers;
-					yield serversDB.action(async() => {
-						const serversCollection = serversDB.collections.get('servers');
-						const allServerRecords = await serversCollection.query().fetch();
-
-						// filter servers
-						let serversToCreate = servers.filter(i1 => !allServerRecords.find(i2 => i1.id === i2.id));
-
-						// Create
-						serversToCreate = serversToCreate.map(record => serversCollection.prepareCreate(protectedFunction((s) => {
-							s._raw = sanitizedRaw({ id: record.id }, serversCollection.schema);
-							Object.assign(s, record);
-						})));
-
-						const allRecords = serversToCreate;
-
-						try {
-							await serversDB.batch(...allRecords);
-						} catch (e) {
-							log(e);
-						}
-						return allRecords.length;
-					});
-				} catch (e) {
-					log(e);
-				}
-			}
-
-			try {
-				yield AsyncStorage.setItem('hasMigration', '1');
-			} catch (e) {
-				log(e);
-			}
-		}
-
 		if (!token || !server) {
 			yield all([
```

Comment:
MMKV is throwing an error if you try to remove something that isn't there.

#### Steps:
- open a fresh app
- go to open
- close and open again
- it crashes

![image](https://user-images.githubusercontent.com/804994/88217789-6e0e8600-cc35-11ea-88ac-5c91bacd772a.png)

Reply:
I made some changes to the `react-native-mmkv-storage` that fixes this, I've tested on 4 different simulators using this same steps and it's not crashing anymore.
Let's see if it's working fine when you test, if yes, I'll open a PR to `react-native-mmkv-storage`. :)
`react-native-mmkv-storage` have an indexer, and it has an array with all the keys of each type (string/number/map), sometimes it's doing something like that:
#### Pseudo code
```haxe
let a = [1];
a.pop();
mmkv.save(a);
```
and it's crashing mmkv since we're trying to save a null/empty array, and it thrown a lot of errors like that you have found.